### ISSUE TITLE
Is this project dead?

### ISSUE THREAD

**Issue by phpfui** [propelorm/Propel2#2014]
Lots of PRs with no comments or merging. Open issues for bugs and even fatal errors.  See https://github.com/propelorm/Propel2/issues/2009

Looks like the old Propel has not been updated for 5 years.

Any insights?

**Comment by DarkAxi0m** [comment]
I would not say it's dead,  maybe not as active as it once was.
There was a realise back in March

**Comment by phpfui** [comment]
It seems like the original creators have moved on, so not sure who is actively in charge. I looked into the issue I found (static call of member method in delete) but did not see an easy fix. This is kind of basic functionality that does not work, so I wondered what is going on with the project.

My interest is benchmarking the package, so not using it in prod.  More curious than anything else. It is sad we start stuck with slow and bloated ORMs like Eloquent.

**Comment by larryzoumas** [comment]
I love Propel but I am using it in production and the writing seems to be on the wall that further updates to PHP core will render it kind of obsolete without a major rewrite. 

What is the closest thing out there that is maintained? Anybody ever use **doctrine-active-record**? 
(https://github.com/symlex/doctrine-active-record)

**Comment by phpfui** [comment]
I had the same problem and one reason I did not consider Propel (but it was closer than most to what I wanted) as it looked like a dying project. I searched far and wide and found nothing to meet my needs (fast and low memory footprint), so I wrote my own: https://github.com/phpfui/ORM  You can check out some of the philosophy behind it on my blog starting here: https://blog.phpfui.com/getters-and-setters-vs-public-access

Check it out to see if it meets your needs.  A slightly different approach than the memory hogging repository models you see most ORMs taking.

**Comment by larryzoumas** [comment]
omg  phpfui. I don't even know what to say...Uhm, thank you? Will definitely check it out.

**Comment by phpfui** [comment]
Let me know if you find any issues. I think it is fresh take on ORMs.  Active Table and database cursors are key to reduced memory usage.

**Comment by fsmeier** [comment]
Heyhey, 
very good question :) 

This project is not dead - yet, it is lacking active maintainers and moderators. 

There is a very small group of people taking care of security issues and php updates because of business needs. But sadly currently no further resources to advance the project. So help is very welcome!

All the best,
Florian

**Comment by phpfui** [comment]
I agree that it was one of the better ORMs out there. But right now you can't delete an Active Record.  Also it has a lot of boiler plate to set up and does not really allow for easy updates of source.

I solved all these issues in my ORM.  I am trying to benchmark Propel2, but it currently does not work enough to benchmark it.

**Comment by markovicdenis** [comment]
> I agree that it was one of the better ORMs out there. But right now you can't delete an Active Record. Also it has a lot of boiler plate to set up and does not really allow for easy updates of source.

It has very powerful generator classes whcih can be extended. You can add any method you want in the generated queries, models, tablemaps etc. Specially a static delete method. That would be so easy to add.

**Comment by phpfui** [comment]
It is overly complex (like most PHP ORMs) and broken unfortunately. And no one is maintaining it.

**Comment by oojacoboo** [comment]
In reality Active Record is a broken pattern.  We migrate to Doctrine a while back and are much happier.  There is a reason it's not maintained - everyone moved to DataMapper and Doctrine years ago.  It started with Symfony switching to Doctrine as the primary ORM, away from Propel.  That was the beginning of the end.

**Comment by markovicdenis** [comment]
> In reality Active Record is a broken pattern. We migrated to Doctrine (partially) a while back and are much happier. There is a reason it's not maintained - everyone moved to DataMapper and Doctrine years ago. It started with Symfony switching to Doctrine as the primary ORM, away from Propel. That was the beginning of the end.

Luckily PHP ads features very slowly so we can still keep using Propel for then next decade 🤣 With a few fixes here and there. Either way, PHP itself is a broken pattern. This is the last time I do anyhthing with it. I am switching to c# or something else.

**Comment by oojacoboo** [comment]
Nothing wrong with PHP these days.  But if you're looking at this Propel repo as an example of what PHP is today, you're getting a really distorted view of reality.

**Comment by markovicdenis** [comment]
> Nothing wrong with PHP these days. But if you're looking at this Propel repo as an example of what PHP is today, you're getting a really distorted view of reality.

I know what PHP is. And given the chance to start the project over, I'd chose C# 10/10 tiems, since I wouldn't have to worry about Microsoft dropping support ofr EF core anytime soon and switching it for soemthing else. Also, C# is miles ahead in terms of language features.

**Comment by phpfui** [comment]
I can understand C# users. A decent language now, but was Windows only for a long time. Also not a fan of Microsoft's constantly changing interfaces. Seems you only get about 5 years out of something before they change to some new and "improved" way that is completely incompatible with your years of effort.

I started this issue asking about the status of this library, and now I know it is pretty dead. Thanks for the insight.

But I will say PHP is easy to work with, but I tend to write code that looks like C++, so I don't see most of the issues people have with PHP written by newbies who think JavaScript is a good language.

**Comment by netbrothers-tr** [comment]
We have migrated multiple projects (different frameworks) from Propel to Doctrine. There has been no problem at all. IMO picking on PHP is out of place here, @markovicdenis, because you can definitely create modern software with PHP that runs virtually everywhere. But it's a good thing, you feel comfortable and satisfied with C#.

**Comment by larryzoumas** [comment]
Hi guys! I just want to thank the original Propel team if they are reading this. It was a truly spectacular piece of software in its time. Truly inspirational.

I am just now migrating a large project from Propel to Doctrine because it seems like there is no upgrade path beyond PHP 8.2 and I want to stay current. The experience has been time-consuming but AI helps... a lot. I would encourage others to do the same and get on with your lives. You can't live in the past!

And finally... whether you code C#, php, python, or whatever... just remember: **Star Trek II: The Wrath of Khan** was the best film ever.

**Comment by smhg** [comment]
> I am just now migrating a large project from Propel to Doctrine because it seems like there is no upgrade path beyond PHP 8.2 and I want to stay current.

I am using Propel **2.0.0-beta4** on PHP 8.3 (with Symfony 7.1) and it works just fine for me. What doesn't work for you?

**Comment by larryzoumas** [comment]
"propel/propel": "2.0.x-dev",

If I remember correctly,l lots of deprecation warnings in 8.3 and 8.4.  I
can't turn those off, at least not in my mind :)

----------------------------------
*Larry Zoumas*
*Stockholm, SWEDEN*

Mobile: +46 70 926 6949
Web: zoumas.net <https://www.zoumas.net>
Email: ***@***.***

**Comment by smhg** [comment]
@larryzoumas a fix for this was merged into master yesterday. Hopefully a new beta release can soon follow!

**Comment by oojacoboo** [comment]
See #2025

**Comment by PhilinTv** [comment]
No, the project is not dead. It's used in many production projects, that assumes enterprise level security and performance, that need to be ensured before we move forward. 

Once we stabilize the current version (fix dependencies, tests) we can tag a major release.

You may also check this discussion here for more details and options: https://github.com/propelorm/Propel2/issues/2025
